### PR TITLE
schema for external list test reports api

### PR DESCRIPTION
**Pull request by Germandrummer92**
Depends on https://github.com/OctoMind-dev/automagically/pull/5011

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -285,10 +330,14 @@ components:
           example: "https://en.wikipedia.org/"
         status:
           type: string
-          enum: ["WAITING", "PASSED", "FAILING"]
+          enum: [ "WAITING", "PASSED", "FAILED" ]
```
Comment (Germandrummer92) [inline openapi.yaml:333]: previously a typo 😬

File: openapi.yaml [hunk 2]
```diff
@@ -285,10 +330,14 @@ components:
...
           description: The status of the test report, will be WAITING as long as any result is running, FAILED if the report is done but has any failed result and PASSED if all test results are done and successful
         context:
           $ref: "#/components/schemas/TestReportContext"
+        testResults:
```
Comment (Germandrummer92) [inline openapi.yaml:337]: they are always included, I double checked all 3 routes

File: openapi.yaml [hunk 3]
```diff
@@ -394,10 +443,85 @@ paths:
...
+                format: date-time
+                description: The timestamp of the key of the next page to fetch - See [Keyset Pagination](https://use-the-index-luke.com/no-offset)
+                example: "2024-09-06T13:01:51.686Z"
+        - in: query
+          name: filter
+          required: false
+          schema:
+            type: object
+            properties:
+              environmentId:
```
Comment (Germandrummer92) [inline openapi.yaml:475]: theoretically this allows OTHER filters as well dynamically based on the type of the testReport -> I think the only one that might make sense for now is the environmentId, so I would only add more here/make it dynamic in the docs if they are requested

**Review by sker65 — APPROVED** [review]

**Review by kosta — COMMENTED** [review]
Did we generate this by hand?

**Comment by Germandrummer92** [comment]
> Did we generate this by hand?

yes. Unless you have some plug-n-play thing of how to generate from next.js route handlers :(

But I have the same concerns as you - it's getting too large to manually maintain at this point unfortunately

**Pull request merged**